src/scripts/docker_migration_apt.sh

```diff
@@ -0,0 +1,168 @@
+  VERSION=$DISTRIB_RELEASE
+elif [ -f /etc/debian_version ]; then
+  OS=Debian
+  VERSION=$(cat /etc/debian_version)
+else
+  OS=$(uname -s)
+  VERSION=$(uname -r)
+fi
+
+if echo "$OS" | grep -Ei "(Debian)" >/dev/null 2>&1; then
+  DOWNLOAD_GPG_URL="https://download.docker.com/linux/debian/gpg"
+  DOWNLOAD_REPO_URL="https://download.docker.com/linux/debian"
+elif echo "$OS" | grep -Ei "(Ubuntu)" >/dev/null 2>&1; then
+  DOWNLOAD_GPG_URL="https://download.docker.com/linux/ubuntu/gpg"
+  DOWNLOAD_REPO_URL="https://download.docker.com/linux/ubuntu"
+elif echo "$OS" | grep -Ei "(Raspbian)" >/dev/null 2>&1; then
+  DOWNLOAD_GPG_URL="https://download.docker.com/linux/raspbian/gpg"
+  DOWNLOAD_REPO_URL="https://download.docker.com/linux/raspbian"
+else
+  log "OS $OS is not supported, skipping upgrade"
+  exit 0
+fi
+
+# Print OS and version for the migration
+log "OS: $OS ; Version: $VERSION ; Docker version: $DOCKER_VERSION"
+
+# Remove legacy docker packages
+for pkg in docker.io docker-doc docker-compose podman-docker containerd runc; do
+  log "Removing $pkg"
+  apt-get remove $pkg 2>&1 | tee -a ${LOG_FILE}
+done
+
+# Set up the repository
+log "Set up the repository"
+# 1. Update the apt package index and install packages to allow apt to use a repository over HTTPS
+log "Update the apt packages"
+apt-get update 2>&1 | tee -a ${LOG_FILE}
+if [ $? -ne 0 ]; then
+  log "Failed to update"
+  exit 1
+fi
+log "Install ca-certificates curl and gnupg"
+apt-get install -y ca-certificates curl gnupg 2>&1 | tee -a ${LOG_FILE}
+if [ $? -ne 0 ]; then
+  log "Failed to install ca-certofocates curl and gnupg."
+  exit 1
+fi
+# 2. Add Docker's official GPG key
+log "Add Docker's official GPG key"
+install -m 0755 -d /etc/apt/keyrings 2>&1 | tee -a ${LOG_FILE}
+if [ $? -ne 0 ]; then
+  log "Failed to create /etc/apt/keyrings directory."
+  exit 1
+fi
+log "Download and install docker gpg key"
+curl -fL "${DOWNLOAD_GPG_URL}" |
+  gpg --yes --dearmor -o /etc/apt/keyrings/docker.gpg 2>&1 | tee -a ${LOG_FILE}
+if [ $? -ne 0 ]; then
+  log "Failed to download and install docker gpg key."
+  exit 1
+fi
+log "Change permissions for docker gpg key"
+chmod a+r /etc/apt/keyrings/docker.gpg 2>&1 | tee -a ${LOG_FILE}
+if [ $? -ne 0 ]; then
+  log "Failed to change permissions for docker gpg key."
+  exit 1
+fi
+# 3. Use the following command to set up the repository
+log "Add docker repository"
+echo \
+  "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] ${DOWNLOAD_REPO_URL} \
+  $(source /etc/os-release && echo "$VERSION_CODENAME") stable" 2>&1 |
+  tee /etc/apt/sources.list.d/docker.list >/dev/null
+if [ $? -ne 0 ]; then
+  log "Failed to add docker repository."
+  exit 1
+fi
+
+# 1. Update the apt package index:
+log "Update the apt packages again"
+apt-get update 2>&1 | tee -a ${LOG_FILE}
+if [ $? -ne 0 ]; then
+  log "Failed to update"
+  exit 1
+fi
+# IMPORTANT: This step MUST be skipped so unattended-upgrades will upgrade docker later on
```

Comment:
We must be sure that the unattended upgrades will install it. Maybe if it is not installed the upgrade is skipped

Reply:
you can verify it with `unattended-upgrade -d`